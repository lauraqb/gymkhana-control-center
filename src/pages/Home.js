import React from 'react';
import PartidasContainer from '../components/Partidas'
import socketIOClient from "socket.io-client";

const serverEndpoint = process.env.REACT_APP_SERVER_ENDPOINT
const socket = socketIOClient(serverEndpoint);

class Home extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            error: null,
            envMissing: false
          };
    }
    componentDidMount() {
        this.setState({ isLoading: true });
        if (serverEndpoint) fetch(serverEndpoint).catch(error => {debugger});
        else this.setState({ envMissing: true });
      }
    
    render() {
        const { envMissing, error } = this.state;
        if (envMissing) {
            return <p>Falta el fichero .env</p>;
        }
        else if (error) {
            return <p>Error del servidor: {error.message}</p>;
        }
        return <PartidasContainer socket={socket}/>
    }
}

export default Home