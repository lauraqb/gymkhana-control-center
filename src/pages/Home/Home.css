.g-title{
    text-align: center;
    margin-top: 10px;
}

/**PartidasList component*/
.nueva-partida-form, .nueva-partida-form .form-control {
    font-size: 0.9rem;;
}

.form-equipo {
    border: 1px solid rgb(187, 187, 187);
    border-radius: 5px;
    padding: 10px 0;
}

a.g-link {
    color: rgb(36, 36, 36);
    text-decoration: none;
}

.icono {
    float: left;
    margin-right: 10px;
}

