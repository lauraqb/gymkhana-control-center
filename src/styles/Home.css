.container-home {
    background-color: #f1ecec;
    height: 100vh;
}


