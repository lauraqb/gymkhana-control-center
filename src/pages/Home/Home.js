import React from 'react';
import GamesList from './components/GamesList'
import Container from 'react-bootstrap/Container'
import './Home.css'
const serverEndpoint = process.env.REACT_APP_SERVER_ENDPOINT

class Home extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            envMissing: false
        }
    }
    componentDidMount() {
        this.setState({ isLoading: true })
        if (!serverEndpoint) {
            this.setState({ envMissing: true })
        }
    }
    
    render() {
        const { envMissing } = this.state;
        return (
            <div className="App-content">
                {envMissing && <p>Falta la variable de entorno REACT_APP_SERVER_ENDPOINT</p>}
                <Container>
                    <div className="g-title"><h3>Centro de Control</h3></div>
                    <GamesList/>
                </Container>
            </div>
        )
    }
}

export default Home