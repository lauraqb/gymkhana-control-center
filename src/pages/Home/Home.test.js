import React from 'react'
import Home from './Home'
import { shallow } from 'enzyme'

describe('Home Page', () => {
    it ('Renders correctly', () => {
        const component = shallow(<Home/>)
        expect(component).toMatchSnapshot()
    })
})