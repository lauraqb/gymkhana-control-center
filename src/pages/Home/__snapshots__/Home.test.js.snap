// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Renders correctly 1`] = `
<div
  className="App-content"
>
  <Container
    fluid={false}
  >
    <div
      className="g-title"
    >
      <h3>
        Centro de Control
      </h3>
    </div>
    <GamesList />
  </Container>
</div>
`;
